07.ls_object/ls.rb

```diff
@@ -18,25 +18,27 @@ def parse_option
   options
 end
 
-def fetch_file_details(file_names, directory)
+def create_file_details(file_names)
   file_names.map do |name|
-    FileDetail.new(name, directory)
+    path = Pathname.new(Dir.pwd).join(name).to_s
+    FileDetail.new(path)
```

Comment:
```suggestion
    FileDetail.new(name)
```

でも問題なく動いたりしませんか？（Rubyのファイル関連のAPIはデフォルトでカレントディレクトリを起点にして動くことが多いような）

Reply:
問題なく動きました！
ファイル名を直接渡すようにします。